// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/actor-tasks/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.methods = exports.properties = exports.rawProperties = exports.name = void 0;
const hooks_1 = require("./hooks");
const runTask = __importStar(require("./run-task"));
const runTaskAndGetDataset = __importStar(require("./run-task-and-get-dataset"));
const operations = [runTask.option, runTaskAndGetDataset.option];
exports.name = 'Actor tasks';
const operationSelect = {
    displayName: 'Operation',
    name: 'operation',
    type: 'options',
    noDataExpression: true,
    displayOptions: {
        show: {
            resource: ['Actor tasks'],
        },
    },
    default: '',
};
operationSelect.options = operations;
operationSelect.default = operations.length > 0 ? operations[0].value : '';
exports.rawProperties = [
    operationSelect,
    ...runTask.properties,
    ...runTaskAndGetDataset.properties,
];
const { properties, methods } = (0, hooks_1.runHooks)(exports.rawProperties);
exports.properties = properties;
exports.methods = methods;
