// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/hooks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.runHooks = runHooks;
const actorResourceLocator_1 = require("./actorResourceLocator");
const actorTaskResourceLocator_1 = require("./actorTaskResourceLocator");
const userActorResourceLocator_1 = require("./userActorResourceLocator");
const runResourceLocator_1 = require("./runResourceLocator");
const keyValueStoreResourceLocator_1 = require("./keyValueStoreResourceLocator");
const keyValueStoreRecordKeyResourceLocator_1 = require("./keyValueStoreRecordKeyResourceLocator");
const genericFunctions_1 = require("./genericFunctions");
function runHooks(properties) {
    const processProperties = (0, genericFunctions_1.compose)(actorResourceLocator_1.overrideActorProperties, actorTaskResourceLocator_1.overrideActorTaskProperties, runResourceLocator_1.overrideRunProperties, keyValueStoreResourceLocator_1.overrideKeyValueStoreProperties, keyValueStoreRecordKeyResourceLocator_1.overrideKeyValueStoreRecordKeyProperties, userActorResourceLocator_1.overrideUserActorProperties);
    return {
        properties: processProperties(properties),
        methods: {
            listSearch: {
                listActors: actorResourceLocator_1.listActors,
                listActorTasks: actorTaskResourceLocator_1.listActorTasks,
                listRuns: runResourceLocator_1.listRuns,
                listKeyValueStores: keyValueStoreResourceLocator_1.listKeyValueStores,
                listKeyValueStoreRecordKeys: keyValueStoreRecordKeyResourceLocator_1.listKeyValueStoreRecordKeys,
                listUserActors: userActorResourceLocator_1.listUserActors,
            },
        },
    };
}
