// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.methods = exports.properties = void 0;
const methods_1 = require("../helpers/methods");
const hooks_1 = require("./hooks");
const actors = __importStar(require("./actors"));
const actorTasks = __importStar(require("./actor-tasks"));
const actorRuns = __importStar(require("./actor-runs"));
const datasets = __importStar(require("./datasets"));
const keyValueStores = __importStar(require("./key-value-stores"));
const authenticationProperties = [];
const resourceSelect = [
    {
        displayName: 'Resource',
        name: 'resource',
        type: 'options',
        noDataExpression: true,
        options: [
            {
                name: 'Actor',
                value: 'Actors',
            },
            {
                name: 'Actor Task',
                value: 'Actor tasks',
            },
            {
                name: 'Actor Run',
                value: 'Actor runs',
            },
            {
                name: 'Dataset',
                value: 'Datasets',
            },
            {
                name: 'Key-Value Store',
                value: 'Key-Value Stores',
            },
        ],
        default: 'Actors',
    },
];
const rawProperties = [
    ...authenticationProperties,
    ...resourceSelect,
    ...actors.properties,
    ...actorTasks.properties,
    ...actorRuns.properties,
    ...datasets.properties,
    ...keyValueStores.properties,
];
const { properties, methods: selfMethods } = (0, hooks_1.runHooks)(rawProperties);
exports.properties = properties;
const methods = (0, methods_1.aggregateNodeMethods)([
    selfMethods,
    actors.methods,
    actorTasks.methods,
    actorRuns.methods,
    datasets.methods,
    keyValueStores.methods,
]);
exports.methods = methods;
