// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/keyValueStoreResourceLocator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.overrideKeyValueStoreProperties = overrideKeyValueStoreProperties;
exports.listKeyValueStores = listKeyValueStores;
const genericFunctions_1 = require("./genericFunctions");
const resourceLocatorProperty = {
    displayName: 'Key-Value Store ID',
    name: 'storeId',
    type: 'resourceLocator',
    default: { mode: 'list', value: '' },
    modes: [
        {
            displayName: 'From list',
            name: 'list',
            type: 'list',
            placeholder: 'Choose...',
            typeOptions: {
                searchListMethod: 'listKeyValueStores',
                searchFilterRequired: false,
                searchable: false,
            },
        },
        {
            displayName: 'ID',
            name: 'id',
            type: 'string',
            validation: [
                {
                    type: 'regex',
                    properties: {
                        regex: '[a-zA-Z0-9]+',
                        errorMessage: 'Not a valid Key-Value Store ID',
                    },
                },
            ],
            placeholder: 'dmXls2mjfQVdzfrC6',
            url: '=https://console.apify.com/storage/key-value-stores/{{ $value }}',
        },
    ],
};
function mapProperty(property) {
    return {
        ...property,
        ...resourceLocatorProperty,
    };
}
function overrideKeyValueStoreProperties(properties) {
    return properties.map((property) => {
        if (property.name === 'storeId') {
            return mapProperty(property);
        }
        return property;
    });
}
async function listKeyValueStores() {
    const searchResults = await genericFunctions_1.apiRequestAllItems.call(this, {
        method: 'GET',
        uri: '/v2/key-value-stores',
        qs: {
            limit: 100,
            offset: 0,
        },
    });
    const { data: { items }, } = searchResults;
    return {
        results: items.map((b) => ({
            name: b.name,
            value: b.id,
            url: `https://console.apify.com/storage/key-value-stores/${b.id}`,
            description: b.name,
        })),
    };
}
