// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/actor-tasks/run-task/properties.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.properties = void 0;
const helpers = __importStar(require("../../../helpers"));
exports.properties = [
    {
        displayName: 'Actor Task',
        name: 'actorTaskId',
        required: true,
        description: 'Task ID or a tilde-separated username and task name',
        default: 'janedoe~my-task',
        type: 'string',
        displayOptions: {
            show: {
                resource: ['Actor tasks'],
                operation: ['Run task'],
            },
        },
    },
    {
        displayName: 'Use Custom Body',
        name: 'useCustomBody',
        type: 'boolean',
        description: 'Whether to use a custom body',
        default: false,
        displayOptions: {
            show: {
                resource: ['Actor tasks'],
                operation: ['Run task'],
            },
        },
    },
    {
        displayName: 'Input (JSON)',
        name: 'customBody',
        type: 'json',
        default: '{}',
        description: 'Custom body to send',
        displayOptions: {
            show: {
                useCustomBody: [true],
                resource: ['Actor tasks'],
                operation: ['Run task'],
            },
        },
    },
    {
        displayName: 'Wait for Finish',
        name: 'waitForFinish',
        description: 'Whether or not to wait for the run to finish before continuing. If true, the node will wait for the run to complete (successfully or not) before moving to the next node. Note: The maximum time the workflow will wait is limited by the workflow timeout setting in your n8n configuration.',
        default: true,
        type: 'boolean',
        displayOptions: {
            show: {
                resource: ['Actor tasks'],
                operation: ['Run task'],
            },
        },
    },
    {
        displayName: 'Timeout',
        name: 'timeout',
        description: `Optional timeout for the run, in seconds. By default, the run uses a
timeout specified in the task settings.`,
        default: null,
        type: 'number',
        displayOptions: {
            show: {
                resource: ['Actor tasks'],
                operation: ['Run task'],
            },
        },
    },
    {
        displayName: 'Memory',
        name: 'memory',
        description: 'Memory limit for the run, in megabytes. The amount of memory can be set to one of the available options. By default, the run uses a memory limit specified in the task settings.',
        default: 1024,
        type: 'options',
        options: helpers.consts.memoryOptions,
        displayOptions: {
            show: {
                resource: ['Actor tasks'],
                operation: ['Run task'],
            },
        },
    },
    {
        displayName: 'Build',
        name: 'build',
        description: `Specifies the Actor build to run. It can be either a build tag or build
number. By default, the run uses the build specified in the task
settings (typically \`latest\`).`,
        default: '',
        type: 'string',
        displayOptions: {
            show: {
                resource: ['Actor tasks'],
                operation: ['Run task'],
            },
        },
    },
];
