// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/actors/scrape-single-url/properties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.properties = void 0;
exports.properties = [
    {
        displayName: 'URL',
        name: 'url',
        description: 'URL to be scraped. Must start with http:// or https:// and be a valid URL.',
        default: 'https://docs.apify.com/academy/web-scraping-for-beginners',
        type: 'string',
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Scrape single URL'],
            },
        },
    },
    {
        displayName: 'Crawler Type',
        name: 'crawlerType',
        default: 'cheerio',
        type: 'options',
        options: [
            {
                name: 'Cheerio',
                value: 'cheerio',
            },
            {
                name: 'JSDOM',
                value: 'jsdom',
            },
            {
                name: 'Playwright Adaptive',
                value: 'playwright:adaptive',
            },
            {
                name: 'Playwright Firefox',
                value: 'playwright:firefox',
            },
        ],
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Scrape single URL'],
            },
        },
    },
];
