// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/actors/run-actor-and-get-dataset/execute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.runActorAndGetDataset = runActorAndGetDataset;
const n8n_workflow_1 = require("n8n-workflow");
const genericFunctions_1 = require("../../genericFunctions");
const executeActor_1 = require("../../executeActor");
async function runActorAndGetDataset(i) {
    const actorId = this.getNodeParameter('actorId', i, undefined, {
        extractValue: true,
    });
    const timeout = this.getNodeParameter('timeout', i);
    const memory = this.getNodeParameter('memory', i);
    const buildParam = this.getNodeParameter('build', i);
    const rawStringifiedInput = this.getNodeParameter('customBody', i, '{}');
    const { runId, lastRunData } = await executeActor_1.executeActor.call(this, {
        actorId,
        timeout,
        memory,
        buildParam,
        rawStringifiedInput,
        waitForFinish: true,
    });
    if (!(lastRunData === null || lastRunData === void 0 ? void 0 : lastRunData.defaultDatasetId)) {
        throw new n8n_workflow_1.NodeApiError(this.getNode(), {
            message: `Run ${runId} did not create a dataset`,
        });
    }
    if ((lastRunData === null || lastRunData === void 0 ? void 0 : lastRunData.status) !== 'SUCCEEDED') {
        throw new n8n_workflow_1.NodeApiError(this.getNode(), {
            message: `Run ${runId} did not finish with status SUCCEEDED. Run status: ${lastRunData === null || lastRunData === void 0 ? void 0 : lastRunData.status}`,
        });
    }
    const datasetItems = await genericFunctions_1.apiRequest.call(this, {
        method: 'GET',
        uri: `/v2/datasets/${lastRunData.defaultDatasetId}/items`,
        qs: { format: 'json' },
    });
    return this.helpers.returnJsonArray(datasetItems);
}
