// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/actors/run-actor/properties.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.properties = void 0;
const helpers = __importStar(require("../../../helpers"));
exports.properties = [
    {
        displayName: 'Actor Source',
        name: 'actorSource',
        type: 'hidden',
        default: 'recentlyUsed',
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Run actor'],
            },
        },
    },
    {
        displayName: 'Actor',
        name: 'actorId',
        required: true,
        description: 'Actor ID or a tilde-separated username and Actor name',
        default: 'janedoe~my-actor',
        type: 'string',
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Run actor'],
            },
        },
    },
    {
        displayName: 'Input JSON',
        name: 'customBody',
        type: 'json',
        default: '{}',
        description: 'JSON input for the Actor run, which you can find on the Actor input page in Apify Console. If empty, the run uses the input specified in the default run configuration. https://console.apify.com',
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Run actor'],
            },
        },
    },
    {
        displayName: 'Wait for Finish',
        name: 'waitForFinish',
        description: 'Whether or not to wait for the run to finish before continuing. If true, the node will wait for the run to complete (successfully or not) before moving to the next node. Note: The maximum time the workflow will wait is limited by the workflow timeout setting in your n8n configuration.',
        default: true,
        type: 'boolean',
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Run actor'],
            },
        },
    },
    {
        displayName: 'Timeout',
        name: 'timeout',
        description: `Optional timeout for the run, in seconds. By default, the run uses a
timeout specified in the default run configuration for the Actor.`,
        default: null,
        type: 'number',
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Run actor'],
            },
        },
    },
    {
        displayName: 'Memory',
        name: 'memory',
        description: 'Memory limit for the run, in megabytes. The amount of memory can be set to one of the available options. By default, the run uses a memory limit specified in the default run configuration for the Actor.',
        default: 1024,
        type: 'options',
        options: helpers.consts.memoryOptions,
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Run actor'],
            },
        },
    },
    {
        displayName: 'Build Tag',
        name: 'build',
        description: `Specifies the Actor build tag to run. By default, the run uses the build specified in the default run
configuration for the Actor (typically \`latest\`).`,
        default: '',
        type: 'string',
        displayOptions: {
            show: {
                resource: ['Actors'],
                operation: ['Run actor'],
            },
        },
    },
];
