// @apify/n8n-nodes-apify/dist/nodes/Apify/Apify.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Apify = void 0;
const Apify_properties_1 = require("./Apify.properties");
const Apify_methods_1 = require("./Apify.methods");
const router_1 = require("./resources/router");
class Apify {
    constructor() {
        this.description = {
            displayName: 'Apify',
            name: 'apify',
            icon: 'file:apify.svg',
            group: ['transform'],
            version: 1,
            subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
            description: 'Access Apify tools for web scraping, data extraction, and automation.',
            defaults: {
                name: 'Apify',
            },
            inputs: ["main"],
            outputs: ["main"],
            usableAsTool: true,
            credentials: [
                {
                    displayName: 'Apify API key connection',
                    name: 'apifyApi',
                    required: false,
                    displayOptions: {
                        show: {
                            authentication: ['apifyApi'],
                        },
                    },
                },
                {
                    displayName: 'Apify OAuth2 connection',
                    name: 'apifyOAuth2Api',
                    required: false,
                    displayOptions: {
                        show: {
                            authentication: ['apifyOAuth2Api'],
                        },
                    },
                },
            ],
            properties: Apify_properties_1.properties,
        };
        this.methods = Apify_methods_1.methods;
    }
    async execute() {
        const items = this.getInputData();
        const returnData = [];
        for (let i = 0; i < items.length; i++) {
            const data = await router_1.resourceRouter.call(this, i);
            if (Array.isArray(data)) {
                returnData.push(...data);
            }
            else {
                returnData.push(data);
            }
        }
        return [returnData];
    }
}
exports.Apify = Apify;
