// @apify/n8n-nodes-apify/dist/nodes/Apify/resources/key-value-stores/get-key-value-store-record/execute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getKeyValueStoreRecord = getKeyValueStoreRecord;
const n8n_workflow_1 = require("n8n-workflow");
const helpers_1 = require("../../../helpers");
const genericFunctions_1 = require("../../genericFunctions");
async function getKeyValueStoreRecord(i) {
    const storeId = this.getNodeParameter('storeId', i);
    const recordKey = this.getNodeParameter('recordKey', i);
    if (!storeId || !recordKey) {
        throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'Store ID and Record Key are required');
    }
    try {
        const apiCallFn = () => this.helpers.httpRequestWithAuthentication.call(this, 'apifyApi', {
            method: 'GET',
            url: `${helpers_1.consts.APIFY_API_URL}/v2/key-value-stores/${storeId.value}/records/${recordKey.value}`,
            headers: {
                'x-apify-integration-platform': 'n8n',
            },
            returnFullResponse: true,
            encoding: 'arraybuffer',
        });
        const apiResult = await (0, genericFunctions_1.retryWithExponentialBackoff)(apiCallFn);
        if (!apiResult) {
            return { json: {} };
        }
        const contentType = apiResult.headers['content-type'];
        const value = apiResult.body;
        const resultBase = {
            storeId: storeId.value,
            recordKey: recordKey.value,
            contentType,
        };
        if (contentType &&
            !contentType.startsWith('application/json') &&
            !contentType.startsWith('text/')) {
            const fileName = recordKey.value || apiResult.key || 'file';
            const binaryData = await this.helpers.prepareBinaryData(value, fileName, contentType);
            return {
                json: { ...resultBase },
                binary: { data: binaryData },
            };
        }
        const buffer = value;
        let finalData;
        if (contentType && contentType.startsWith('application/json')) {
            try {
                finalData = JSON.parse(buffer.toString('utf8'));
            }
            catch {
                finalData = buffer.toString('utf8');
            }
        }
        else if (contentType && contentType.startsWith('text/')) {
            finalData = buffer.toString('utf8');
        }
        else {
            finalData = buffer.toString('base64');
        }
        return { json: { ...resultBase, data: finalData } };
    }
    catch (error) {
        throw new n8n_workflow_1.NodeApiError(this.getNode(), error);
    }
}
