// @apify/n8n-nodes-apify/dist/nodes/Apify/helpers/consts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEFAULT_EXP_BACKOFF_RETRIES = exports.DEFAULT_EXP_BACKOFF_EXPONENTIAL = exports.DEFAULT_EXP_BACKOFF_INTERVAL = exports.memoryOptions = exports.WAIT_FOR_FINISH_POLL_INTERVAL = exports.TERMINAL_RUN_STATUSES = exports.APIFY_API_URL = exports.WEB_CONTENT_SCRAPER_ACTOR_ID = void 0;
exports.WEB_CONTENT_SCRAPER_ACTOR_ID = 'aYG0l9s7dbB7j3gbS';
exports.APIFY_API_URL = 'https://api.apify.com';
exports.TERMINAL_RUN_STATUSES = ['SUCCEEDED', 'FAILED', 'TIMED-OUT', 'ABORTED'];
exports.WAIT_FOR_FINISH_POLL_INTERVAL = 1000;
exports.memoryOptions = [
    { name: '128 MB', value: 128 },
    { name: '256 MB', value: 256 },
    { name: '512 MB', value: 512 },
    { name: '1024 MB (1 GB)', value: 1024 },
    { name: '2048 MB (2 GB)', value: 2048 },
    { name: '4096 MB (4 GB)', value: 4096 },
    { name: '8192 MB (8 GB)', value: 8192 },
    { name: '16384 MB (16 GB)', value: 16384 },
    { name: '32768 MB (32 GB)', value: 32768 },
];
exports.DEFAULT_EXP_BACKOFF_INTERVAL = 1;
exports.DEFAULT_EXP_BACKOFF_EXPONENTIAL = 2;
exports.DEFAULT_EXP_BACKOFF_RETRIES = 5;
